client/checkout/blocks/upe.js

```diff
@@ -0,0 +1,53 @@
+/**
+ * External dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+// Handled as an external dependency: see '/webpack.config.js:83'
+import {
+	registerPaymentMethod,
+	registerExpressPaymentMethod,
+	// eslint-disable-next-line import/no-unresolved
+} from '@woocommerce/blocks-registry';
+
+/**
+ * Internal dependencies
+ */
+import { PAYMENT_METHOD_NAME_CARD } from '../constants.js';
+import { getConfig } from 'utils/checkout';
+import WCPayAPI from './../api';
+import WCPayUPEFields from './upe-fields.js';
+import request from './request.js';
+import enqueueFraudScripts from 'fraud-scripts';
+import paymentRequestPaymentMethod from '../../payment-request/blocks';
+
+// Create an API object, which will be used throughout the checkout.
+const api = new WCPayAPI(
+	{
+		publishableKey: getConfig( 'publishableKey' ),
+		accountId: getConfig( 'accountId' ),
+		forceNetworkSavedCards: getConfig( 'forceNetworkSavedCards' ),
+		locale: getConfig( 'locale' ),
+		isUPEEnabled: getConfig( 'isUPEEnabled' ),
+	},
+	request
+);
+
+registerPaymentMethod( {
+	name: PAYMENT_METHOD_NAME_CARD,
+	content: <WCPayUPEFields api={ api } />,
+	edit: <WCPayUPEFields api={ api } />,
+	canMakePayment: () => !! api.getStripe(),
+	paymentMethodId: PAYMENT_METHOD_NAME_CARD,
+	label: __( 'WooCommerce Payments', 'woocommerce-payments' ),
+	ariaLabel: __( 'WooCommerce Payments', 'woocommerce-payments' ),
```

Comment:
Is there no way to pull these fields from the Gateway class instead of repeating them here? While I am criticising this, ironically this may actually work out in our favour, if we wanted to keep this title dynamic dependent on enabled payment methods, because I think we should be able to pull enabled payment methods using `getConfig( 'paymentMethodsConfig' )` at this point. Let me know if you prophetically had this in your mind over here!

Reply:
So there was a little bit of foresight going in to this decision as you mentioned since I knew we wanted to alter these titles anyway. I also was just using the existing CC implementation as a starting point and went with their approach. With that in mind do you think it would be best to keep this as is and address the dynamic title in #2433?

Reply:
I think let's leave it like this; I was just thinking out loud with my comment and as I did so, eventually came around to this same reasoning. I think #2433 will probably end up overwriting this: it'll still probably be nice to pull the optional values here from the gateway config, but that can be taken care of properly in the separate issue.